docs/pages/reference/terraform/resources/app.mdx

```diff
@@ -0,0 +1,108 @@
+---
+title: "teleport_app Resource - teleport"
+description: |-
+  
+---
+
+# teleport_app (Resource)
+
+
+
+## Example Usage
+
+```hcl
+# Teleport App
+
+resource "teleport_app" "example" {
+  metadata = {
+    name        = "example"
+    description = "Test app"
+    labels = {
+      "teleport.dev/origin" = "dynamic" // This label is added on Teleport side by default
+    }
+  }
+
+  spec = {
+    uri = "localhost:3000"
+  }
+}
+```
+
+{/*  schema generated by tfplugindocs */}
+## Schema
+
+### Required
+
+- `version` (String) Version is the resource version. It must be specified. Supported values are:`v3`.
+
+### Optional
+
+- `kind` (String) Kind is the app resource kind. Always "app".
+- `metadata` (Attributes) Metadata is the app resource metadata. (see [below for nested schema](#nested-schema-for-metadata))
+- `spec` (Attributes) Spec is the app resource spec. (see [below for nested schema](#nested-schema-for-spec))
+- `sub_kind` (String) SubKind is an optional resource subkind.
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+### Nested Schema for `metadata`
+
+Required:
+
+- `name` (String) Name is an object name
+
+Optional:
+
+- `description` (String) Description is object description
+- `expires` (String) Expires is a global expiry time header can be set on any resource in the system.
+- `labels` (Map of String) Labels is a set of labels
+- `namespace` (String) Namespace is object namespace. The field should be called "namespace" when it returns in Teleport 2.4.
```

Comment:
For the most part, I think feedback on individual field descriptions is outside the scope of my review (since there are so many cosmetic changes we can make to field comments in the source), but we should definitely edit the namespace description, since this description has the potential to be misleading and it's present in every single dynamic resource. 

As far as I remember, I think it's also a deprecated field with `default` as the only permissible value, right?

Reply:
I updated the proto files to mark namespacing as deprecated and that the only supported values are empty or `default`.